compiler/passes/normalize.cpp

```diff
@@ -386,65 +388,27 @@ static void transformLogicalShortCircuit()
   }
 }
 
-
-static Symbol* reduceIntentOp(ForallIntents* fi, Symbol* reVar) {
-  int nv = fi->numVars();
-  for (int i = 0; i < nv; i++)
-    if (fi->isReduce(i))
-      if (SymExpr* varSE = toSymExpr(fi->fiVars[i]))
-        if (varSE->symbol() == reVar) {
-          SymExpr* ri = toSymExpr(fi->riSpecs[i]);
-          INT_ASSERT(ri);
-          return ri->symbol();
-        }
-
-  // Did not see 'reVar' with a reduce intent.
-  return NULL;
-}
-
-//
-// lowerReduceAssign(): lower the reduce= calls, verifying their correct use.
 //
-// Parser represents "x reduce= y" in source code as PRIM_REDUCE_ASSIGN(x,y).
-// lowerReduceAssign() converts it into globalOp.accumulateOntoState(x,y)
-// where globalOp comes from the reduce intent for x.
-// This includes checking that there is, indeed, a reduce intent for x:
-// without it, there is no globalOp.
+// handleReduceAssign(): check+process the reduce= calls
 //
-static void lowerReduceAssign() {
+static void handleReduceAssign() {
```

Comment:
I thought that the new ForallStmt is only used in some cases, and in particular not for forall expressions. Did you change reduce intents for both cases? Or can you not use reduce= for forall expressions?

I see below the error saying it must be in a "forall loop". Does that mean not a forall expression? Because that is also a kind of loop, isn't it?

Should the error say "must be in a forall statement"? And do you expect `reduce=` will ever be supported in forall expressions?